crates/accelerate/src/euler_one_qubit_decomposer.rs

```diff
@@ -688,15 +688,16 @@ fn mod_2pi(angle: f64, atol: f64) -> f64 {
 }
 
 fn params_zyz_inner(mat: ArrayView2<Complex64>) -> [f64; 4] {
-    let coeff: Complex64 = 1. / det_one_qubit(mat).sqrt();
+    let det = det_one_qubit(mat);
+    let coeff: Complex64 = 1. / det.sqrt();
     let phase = -complex_phase(coeff);
     let tmp_1_0 = (coeff * mat[[1, 0]]).abs();
     let tmp_0_0 = (coeff * mat[[0, 0]]).abs();
     let theta = 2. * tmp_1_0.atan2(tmp_0_0);
```

Comment:
I didn't notice this at the time we merged this, but since `tmp_1_0` and `tmp_0_0` are both guaranteed positive real and only used in an `atan2` call, multiplying by `coeff` has no effect (it immediately cancels), and I think this could just be `theta = 2. * mat[[1, 0]].abs().atan2(mat[[0, 0]].abs())`.

Reply:
Come to think of it, if we do that, I don't even think we need to calculate the reciprocal square root, which is actually a quite nice saving.

Reply:
I suppose technically it has an effect if `coeff` is 0 or infinite, but that would require an infinite or zero determinant respectively, which can't happen for unitary matrices.

Reply:
I made these simplification in: https://github.com/Qiskit/qiskit/pull/10789/commits/4c08c0adf46d060c14b581561ad1a51c59199e58

Reply:
since `mat[[1,0]].abs()` and `mat[[0,0]].abs()` are both real, could it make sense to call `atan` instead of `atan2`? Just in case that's any faster

Reply:
`mat[[0, 0]]` could still be zero I think?  We wouldn't be able to produce the argument to pass to `atan` by division if so, in which case we'd just need to do the branching that `atan2` will do anyway.

Reply:
oh right, that makes sense.